### ISSUE TITLE
Bug: Image seem very blurry even though it has very high resolution

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [Yalantis/uCrop#466]
If you take an image with large resolution, and you zoom in, it becomes very blurry, even though it has the needed details.

Take this image for example:
 
[original.zip](https://github.com/Yalantis/uCrop/files/2520753/original.zip)

If you zoom in, you see less details than if you open the same file on a Gallery app.
In fact, in the sample, I can't zoom in enough...

**Comment by p1nkydev** [comment]
@AndroidDeveloperLB Did you use with native implementation?

**Comment by AndroidDeveloperLB** [comment]
@p1nkydev I just tried the sample of the project. I didn't change anything except making it work.

**Comment by p1nkydev** [comment]
@AndroidDeveloperLB Can you provide image you use?

**Comment by AndroidDeveloperLB** [comment]
@p1nkydev I already have. Please re-read what I wrote.

**Comment by p1nkydev** [comment]
@AndroidDeveloperLB I guess the issue is that sample uses non-native version, try to import native library in sample or your working project  by adding dependency implementation 'com.github.yalantis:ucrop:2.2.2-native'

**Comment by AndroidDeveloperLB** [comment]
@p1nkydev This shouldn't be required. There are other libraries that show it's possible to do it without native code. Please try it without native code and you will see the issue.

**Comment by p1nkydev** [comment]
@AndroidDeveloperLB I tried with non-native implementation, seems like it has too much pixels and instead of OutOfMemory you're getting not full image.

**Comment by AndroidDeveloperLB** [comment]
@p1nkydev Why would it get OOM ? Why not downsampling ?

**Comment by p1nkydev** [comment]
@AndroidDeveloperLB You are **not** getting OOM, image is downsampled, that's why quality is a bit worse.

**Comment by AndroidDeveloperLB** [comment]
@p1nkydev That's incorrect to assume that because of downsampling, the part that you view must be in low quality.
You can downsample the part of the image, using BitmapRegionDecoder class, for example:

https://developer.android.com/reference/android/graphics/BitmapRegionDecoder

Also, it's incorrect to assume I will get OOM. On new Android versions ( from Android O), the bitmap is stored on the native RAM, so it's very hard to reach a state that you are out of memory (yet it is still possible, but without this exception) . See here:

https://stackoverflow.com/q/48091403/878126
